Feature/1.0.0/fork

tjbenton opened the pull request:
Resolves #102

tjbenton committed:
updated dependencies

tjbenton committed:
fixed models spinner to be correct

tjbenton committed:
bugfix `parseModelDependencies`

Fixes a bug where if two files required the same dependency the second model passed would have it's dependencies removed which causes issues when trying to make the document generation smarter.

tjbenton committed:
updated to exit if lint fails

tjbenton committed:
bugfix for memory leak

tjbenton committed:
added `resolveDependants`

This adds an array of model files that are dependent on each model

tjbenton committed:
added `events-async`

tjbenton committed:
Created a `Documents` class that runs the models in a more efficient way

bentonam commented on app/models.js:
```diff
@@ -189,9 +187,12 @@ export default class Models extends Base {
     if (this.models.length === this.registered_models.length) {
       this.progress.stop();
+      // unset progress so when it's run again it works correctly
+      // this.progress = null;
```
Remove commented code

bentonam reviewed (approved):
Fix the build and remove commented out code

tjbenton committed:
Merge branch 'release/1.0.0' of https://github.com/bentonam/fakeit into feature/1.0.0/fork

* 'release/1.0.0' of https://github.com/bentonam/fakeit:
  Updated output tests
  Updated cli to show help when no models are passed
  Ensured cli arguments are assigned to variables
  Removed username validation from couchbase
  Fixed Spacing in Model

tjbenton committed:
pr fixes

tjbenton committed:
if the model that's passed is `flight-data` make it run serial

This is to improve performance on all tests since the flight-data tests are the most cpu intensive. It increases the speed of the tests, and decreases the cpu usage

tjbenton committed:
updated to `events-async@1.1.0`

tjbenton committed:
fixed error with `make lint` on node v4
